optimum/onnxruntime/trainer.py

```diff
@@ -542,20 +542,31 @@ def _inner_training_loop(
                 )
 
             self.model = model
-            deepspeed_engine, optimizer, lr_scheduler = deepspeed_init(
-                self, num_training_steps=max_steps, resume_from_checkpoint=resume_from_checkpoint
+            optimizer, lr_scheduler = deepspeed_init(
+                self, num_training_steps=max_steps
             )
-            self.model = unwrap_model(deepspeed_engine)
-            self.model_wrapped = deepspeed_engine
-            self.deepspeed = deepspeed_engine
+            # deepspeed ckpt loading
+            if resume_from_checkpoint is not None and self.is_deepspeed_enabled:
```

Comment:
can we keep the same order of https://github.com/huggingface/transformers/commit/a73b1d59a34ca8e30098b3e32a977a7928663559#diff-ed55888e6665791fe92cc8fc0c499da54f4ace6738551cd9a2591881cda076deR1779 did? 

I mean we don't need put every deepspeed related things in one code block. Following what HF trainer did will make it easier for us to catch up with later HF trainer changes.